test/distributed/org/apache/cassandra/distributed/upgrade/UpgradeTestBase.java

```diff
@@ -196,4 +196,14 @@ public TestCase nodesToUpgrade(int ... nodes)
         }
     }
 
+    protected TestCase allUpgrades(int nodes, int... toUpgrade)
```

Comment:
Nice utility function. Maybe we could have a variant asking for a range of versions, like everything up to 3.x, that could be used in `mixedModeReadRepairCompactStorage`. Not saying that we should do it in this ticket, though.